parse_test.go

```diff
@@ -339,19 +340,25 @@ type ParserData struct {
 	GpgSignature []byte
 }
 
-// filterNondeterministicData replaces some generated fields with static strings
-func filterNondeterministicData(data ParserData) ParserData {
+// filterData replaces some generated fields with static strings
+func filterData(data ParserData) ParserData {
 	copiedData := maps.Clone(data.Metadata)
 
 	fields := []string{
 		"blake2_256_digest",
 		"md5_digest",
 		"sha256_digest",
+		"version",
 	}
 
+	re := regexp.MustCompile(`[a-zA-Z0-9]`)
+
 	for _, field := range fields {
-		if _, ok := copiedData[field]; ok {
-			copiedData[field] = []string{fmt.Sprintf("%s exists", field)}
+		if value, ok := copiedData[field]; ok {
+			values := lo.Map(value, func(item string, _ int) string {
+				return re.ReplaceAllString(item, "X")
+			})
+			copiedData[field] = values
 		}
```

Comment:
Would it be easier to delete the fields from the map entirely?
```suggestion
		if value, ok := copiedData[field]; ok {
			delete(copiedData, field)
		}
```

Reply:
I guess your intent is to try to preserve the information? I'm not sure if it is useful and I think this could still break if the length of the version string changed from something like `1.2.3` to `1.2.3dev0`. Or am I misunderstanding how it replaces?

Reply:
That is how it works.

I was trying to prevent deleting anything since it reduces how effective these tests are, but you are right that this will cause issues for versions.

Reply:
If we wanted to preserve that it is set to _something_, maybe we can just set it to the field name?
```suggestion
		if _, ok := copiedData[field]; ok {
			copiedData[field] = field
		}
```

I'm not sure if that is any better than just deleting the fields that change though. I'm good with whichever you think is best!

Reply:
Yeah, that's what we did before. I think you're right that it's better.